workbench-core/authorization/src/caslAuthorizationPlugin.test.ts

```diff
@@ -154,4 +161,98 @@ describe('CASL Authorization Plugin', () => {
       }
     );
   });
+
+  describe('isAuthorizedOnDynamicOperations', () => {
+    let mockIdentityPermissions: IdentityPermission[];
+    beforeEach(() => {
+      mockIdentityPermissions = [
+        {
+          effect: 'ALLOW',
+          action: 'CREATE',
+          identityType: 'GROUP',
+          identityId: 'SampleGroup',
+          subjectType: 'SampleSubject',
+          subjectId: '*',
+          conditions: { env: { $eq: 'SampleEnv' } }
+        },
+        {
+          effect: 'ALLOW',
+          action: 'UPDATE',
+          identityType: 'GROUP',
+          identityId: 'SampleGroup',
+          subjectType: 'SampleSubject',
+          subjectId: '1234'
+        },
+        {
+          effect: 'DENY',
+          action: 'UPDATE',
+          identityType: 'GROUP',
+          identityId: 'SampleGroup',
+          subjectType: 'SampleSubject',
+          subjectId: '1234'
+        }
+      ];
+      caslAuthorizationPlugin = new CASLAuthorizationPlugin();
+    });
+    test('Check for valid dynamic operations with valid conditions', async () => {
+      const mockDynamicOperations: DynamicOperation[] = [
+        {
+          action: 'CREATE',
+          subject: {
+            subjectType: 'SampleSubject',
+            subjectId: '1',
+            env: 'SampleEnv'
+          }
+        }
+      ];
+      const response = await caslAuthorizationPlugin.isAuthorizedOnDynamicOperations(
+        mockIdentityPermissions,
+        mockDynamicOperations
+      );
+      expect(response).toBeUndefined();
+    });
+
+    test('Check for a invalid dynamic operations due invalid conditions', async () => {
+      const mockDynamicOperations: DynamicOperation[] = [
+        {
+          action: 'CREATE',
+          subject: {
+            subjectType: 'SampleSubject',
+            subjectId: '1'
+          }
+        }
+      ];
+      try {
+        await caslAuthorizationPlugin.isAuthorizedOnDynamicOperations(
+          mockIdentityPermissions,
+          mockDynamicOperations
+        );
+        expect.hasAssertions();
```

Comment:
you could pull `expect.hasAssertions()` out into a `beforeEach` block since every test ran should assert something.